### PR TITLE
feat: parses command line arguments as input WDL values

### LINKED ISSUE #71: feature request: Parsing of command line args as WDL input values
This can be added in a follow-up PR to the `input` module being added in #69 .

`miniwdl` has this feature and it's very useful for `run` commands (which does not yet exist in `sprocket`, but will be added soon). The miniwdl implementation can be reviewed here - https://github.com/chanzuckerberg/miniwdl/blob/main/WDL/CLI.py#L951
miniwdl is able to merge the inputs from a JSON file and the CL, where CL args take precedence and overwrite values present in the JSON. We want to do the same, as that is very handy for scripting.

Primitive type inputs should all be handled in the form `<input_key>=<value>`. Compound types, structs, and objects, are trickier, and I don't yet know what the most ergonomic CLI would be 🤔

### PR DESCRIPTION
**Pull request by DeepakSilaych**
This PR implements the ability to override input values for WDL workflows via command-line arguments. This feature allows users to modify values in their input JSON/YAML files without needing to edit the files directly.

Core Implementation:
- Added overrides.rs for parsing and applying command-line input overrides.
- Introduced overrides_errors.rs for custom error handling related to input overrides.
- Created overrides_tests.rs for unit tests covering various scenarios and edge cases.

Modular Structure:
- Split the original large overrides.rs file into smaller, focused modules for better maintainability.

Error Handling:
- Implemented comprehensive error handling using the thiserror crate.

Documentation:
- Updated docs/input-overrides.md to include usage examples and syntax for the new feature.
- Ensured that the README reflects the new functionality.

Related Issue:
Closes #71 

Checklist:
[x] Added a few sentences describing the PR here.
[x] Added myself as the assignee.
[x] Added relevant code reviewers to the PR.
[x] Code builds clean without any errors or warnings.
[x] Added tests for the new functionality.
[x] Updated the README and documentation to account for these changes.

Please review the changes and provide feedback. Thank you!

### CODE REVIEW

File: src/input/overrides.rs [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+    // Handle arrays with brackets
+    if input.starts_with('[') {
+        return parse_array_value(input);
+    }
+
+    // Handle flat arrays (comma-separated)
+    if input.contains(',') {
+        return parse_flat_array(input);
+    }
```
Comment (a-frantz) [inline src/input/overrides.rs]: I think the API will be a bit cleaner and easier to understand if _all_ arrays must be wrapped in brackets. It should also simplify your code, so it's a win win for everyone!

File: src/input/overrides.rs [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+    // Handle null
+    if input == "null" {
```
Comment (a-frantz) [inline src/input/overrides.rs]: `null` is for JSON and YAML, but in WDL the keyword is `None`. We should treat them equivalently.

File: README.md [hunk 3]
```diff
@@ -72,6 +75,20 @@ Sprocket is available as a Docker [image](https://github.com/stjude-rust-labs/sp
...
+### Input Overrides
+
+Sprocket supports overriding input values via command-line arguments:
+
+```bash
+# Override input values without editing the JSON file
+sprocket validate-inputs workflow.wdl --inputs inputs.json \
+  workflow.sample_name=TUMOR_002 \
+  workflow.threads=8 \
+  workflow.reference=/new/reference.fa
```
Comment (a-frantz) [inline README.md:87]: See how everything is prefix with `workflow.`? That will be annoying to repeatedly type. Instead we should have a `--name` arg which is the name of the workflow _or task_ being run. And then internally, we just prepend every key with `args.name`

File: src/input/mod.rs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+//! Input processing functionality for Sprocket.
+
+pub mod yaml;
+pub mod overrides;
```
Comment (a-frantz) [inline src/input/mod.rs]: I would rename `overrides`. I don't think there's any reason that we haven to _require_ an initial JSON file, and we should instead support specifying every input for the task or workflow via command line. And in that case, "override" stops making sense as the name for this module. I would instead name this `command_line`.

File: src/input/overrides_tests.rs
Comment (a-frantz) [inline src/input/overrides_tests.rs:1]: style nit: we (`stjude-rust-labs`) don't put our tests in a separate file, and instead we stick them at the bottom of the file containing the things being tested. It doesn't make much practical difference one way or the other, but it's good to be consistent within a codebase.

**Review by a-frantz — COMMENTED** [review]
Just some initial "bigger picture" feedback. Overall this is looking sweet! Just suggesting some restructuring and tweaks :)

File: src/input/overrides_tests.rs
Comment (DeepakSilaych) [reply src/input/overrides_tests.rs:1]: I started with a single file, but split it later as the size grew. I’ll recombine it for consistency!

**Comment by a-frantz** [comment]
This is waiting on https://github.com/stjude-rust-labs/wdl/pull/352

Once 352 merges, this should be refactored to use the latest commit on `main` of `wdl`

**Comment by claymcleod** [comment]
Hey @DeepakSilaych, are you still working on this? If not, do you mind if I take your progress from here and add a few things I'd like to do on top of it + merge it in?

**Comment by DeepakSilaych** [comment]
i am waiting for [352](https://github.com/stjude-rust-labs/wdl/pull/352) to merge as it is build on top of that, but if you want to add, go ahead

**Comment by claymcleod** [comment]
That PR is already merged—I'm happy for you to do the rebasing, or I can also do it when I add my few things. Up to you!

File: Cargo.lock [hunk 5]
```diff
@@ -1,3212 +0,0 @@
-# This file is automatically @generated by Cargo.
```
Comment (claymcleod) [inline Cargo.lock:1]: We want to keep the `Cargo.lock` file, as this is primarily a binary (not a library).

**Comment by DeepakSilaych** [comment]
I will rebase it quickly and inform you, then you can add them
